composer/datasets/ade20k_hparams.py

```diff
@@ -75,75 +69,30 @@ def initialize_object(self, batch_size, dataloader_hparams) -> DataSpec:
         self.validate()
 
         if self.use_synthetic:
-            if self.split == 'train':
-                total_dataset_size = 20_206
-            elif self.split == 'val':
-                total_dataset_size = 2_000
-            else:
-                total_dataset_size = 3_352
-
-            dataset = SyntheticBatchPairDataset(
-                total_dataset_size=total_dataset_size,
-                data_shape=[3, self.final_size, self.final_size],
-                label_shape=[self.final_size, self.final_size],
-                num_classes=150,
-                num_unique_samples_to_create=self.synthetic_num_unique_samples,
+            return build_synthetic_ade20k_dataloader(
+                batch_size=batch_size,
+                split=self.split,
+                drop_last=self.drop_last,
+                shuffle=self.shuffle,
+                final_size=self.final_size,
+                num_unique_samples=self.synthetic_num_unique_samples,
                 device=self.synthetic_device,
                 memory_format=self.synthetic_memory_format,
+                **asdict(dataloader_hparams),
```

Comment:
```suggestion
            return build_synthetic_ade20k_dataloader(
                batch_size=batch_size,
                datadir=self.datadir,
                split=self.split,
                drop_last=self.drop_last,
                shuffle=self.shuffle,
                final_size=self.final_size,
                num_unique_samples=self.synthetic_num_unique_samples,
                device=self.synthetic_device,
                memory_format=self.synthetic_memory_format,
                **asdict(dataloader_hparams),
```

Reply:
think you forgot to pass in datadir here and in the synthetic function?

Reply:
ah, synthetic datasets don't have a datadir!